homeassistant/components/auth/__init__.py

```diff
@@ -220,47 +225,55 @@ def __init__(self, retrieve_credentials):
         user = request['hass_user']
 
         credentials = self._retrieve_credentials(
-            data['client_id'], data['code'])
+            data['client_id'], RESULT_TYPE_CREDENTIALS, data['code'])
 
-        if credentials is None:
+        if credentials is None or not isinstance(credentials, Credentials):
```

Comment:
I think this check is no longer necessary given you pass in `RESULT_TYPE_CREDENTIALS`